src/components/organisms/Account.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { FC, useState, useEffect } from 'react'
+import Web3 from 'web3'
+import { makeStyles, Theme } from '@material-ui/core/styles'
+import Popover from '@material-ui/core/Popover'
+import ProviderInfo, { EProvider } from '../../models/ProviderInfo'
+import { shortenString } from '../../utils'
+import { Button, NetworkIndicator, Typography } from '../atoms'
+import { colors, fonts } from '../../theme'
+import { AccountModal, WrongNetworkModal } from '../molecules'
+import ConnectionStatus from '../../models/Enums'
+import { getConnectionStatus } from '../../services/Web3Service'
+import NetworkInfo from '../../models/NetworkInfo'
+
+export interface AccountProps {
+  requiredNetworkName: string
+  setProvider: (provider: EProvider) => Promise<void>
+  web3?: Web3
+  networkInfo?: NetworkInfo
+  account?: string
+  availableProviders?: ProviderInfo[]
+  requiredNetworkId?: number
+  onCorrectNetworkMessage?: string
+  onNetworkMismatchMessage?: string
+  noNetworkMessage?: string
+}
```

Comment:
that's rather a lot of props. can it not be reduced somehow?

Reply:
Actually, looking at how it is used, can the <Web3Provider.Consumer> be created inside the Account component instead?

Reply:
Good point, I will try to combine some of the props :)
About using the Context inside the component, I'm trying not to do that so users can use the library independently of the context

Reply:
Resolving as we agreed to postpone this refactor :)